stdlib/scripts/build-stdlib.sh

```diff
@@ -19,6 +19,17 @@ REPO_ROOT="${SCRIPT_DIR}"/../..
 BUILD_DIR="${REPO_ROOT}"/build
 mkdir -p "${BUILD_DIR}"
 
+ACTUAL_COMPILER_VERSION=$(mojo --version | tr " " "\n" | sed -n 2p)
+EXPECTED_COMPILER_VERSION=$(<"${REPO_ROOT}"/stdlib/COMPATIBLE_COMPILER_VERSION)
+
+if [ "${EXPECTED_COMPILER_VERSION}" != "${ACTUAL_COMPILER_VERSION}" ]; then
+  echo "Mismatch in compiler versions! Cannot build the standard library."
+  echo "Expected compiler version: ${EXPECTED_COMPILER_VERSION}"
+  echo "Current installed compiler version: ${ACTUAL_COMPILER_VERSION}"
+  echo "Please run modular update nightly/mojo to get the latest compiler."
+  exit 1
```

Comment:
**Suggestion** I wonder if this should only be a warning, or if we should include a CLI flag to disable this check. Doing an exact match means this check may have a high false positive rate for incompatibility, which could be a burden as much as it helps.

Reply:
I'd like to do a "min compiler version compatibility" check, but that pushes complexity internally for us to _know_ when there was a breaking change.  We don't have that ability easily (automated) right now other than knowing our internal CI when we test with latest changes internally for the unreleased Mojo SDK combined with the public `mojo` nightly changes fail the CI when you install `nightly/mojo` (the previous released `nightly/mojo` pedantically, by definition).  We can make that better over time.  This whole PR is just a short-term "stop-gap" solution to improve the developer experience in my mind.

Reply:
In the future, the issue will go away anywway no? If the mojo compiler and stdlib are both open source and in the same repo, everything will be in sync and it's going to be rainbows and butterflies :) That's ok if it's a bit rough until then

Reply:
Right, the issue would go away if the compiler and library were both open-source and built in tandem, if you will.  Right now, as I was publishing a new `nightly/mojo` package, there were some breaking changes in the compiler.  The user experience is pretty poor: pages full of cryptic MLIR errors.  😄

Reply:
Since we just want to improve the error message, why do we need the "exit 1"? Would it be possible to just add a big fat warning before or after running the tests? With this, we get a good error message, but it doesn't prevent the tests from actually running if, by luck, everything is compatible, even with a version mismatch (meaning no breaking changes in the compiler)

Reply:
Given that this is a short-term solution, I'm going to stick with the `exit 1`.  You won't even be able to _build_ the new standard library code with the old compiler — much less run the tests.  I think hard-erroring for now is probably going to cause other developers less confusion than trying to understand why they're getting long error messages when trying to build the standard library itself.  These long error messages would make the proposed warning message from this `build-stdlib.sh` script itself, sort of defeating the purpose.